azavea/echo-locator
# Feature/kak/hover interactivity#161

## Overview

Modify map interactivity:

 - make tooltips dynamic (show message in toolitp if neighborhood is not reachable)
 - make map marker green on list card hover (fixes #212)
 - set active neighborhood on map hover in list view, but not detail view (shows route)
 - style reachable neighborhood bounds as green on hover in both list and detail views


### Demo

On list card or neighborhood map bounds hover:
![image](https://user-images.githubusercontent.com/960264/59223334-a79f4180-8b99-11e9-99b6-86a489790685.png)

Hovering over a reachable neighborhood in details view:

![echo_routable_hover](https://user-images.githubusercontent.com/960264/59223534-285e3d80-8b9a-11e9-9a3f-faa9bf5b73f5.png)

Hovering over a non-reachable neighborhood in details view (see tooltip at bottom):

![echo_nonroutable_hover](https://user-images.githubusercontent.com/960264/59223541-2bf1c480-8b9a-11e9-993f-4fa3ee051eaf.png)


### Notes

The end leg and neighborhood centroid end marker were removed from route rendering in #189, as requested by the client. The centroid markers are still present here in list view for the three currently listed neighborhoods, but not in the detail view, and the end leg does not render anywhere.

No modifications are made to map cursor appearance, as Leaflet doesn't support per-feature interactivity (see [this comment](https://github.com/azavea/echo-locator/issues/161#issuecomment-500472444)).

There is a small delay added here to showing the route on hover, as doing so immediately would cause Leaflet event errors as the data updates on the change to the active neighborhood.


## Testing Instructions

 * Map and list card interactions should function as expected
 * Reachable neighborhood bounds should turn green on hover, and in list view, show route
 * Non-reachable neighborhood bounds should only show special tooltip on hover, and not turn green


Closes #161 
Fixes #212
